src/tools/manifest2proto.ts

```diff
@@ -57,23 +58,69 @@ function encodePayload(payload: {}): Uint8Array {
 }
 
 async function particleSpecToProtoPayload(spec: ParticleSpec) {
+  let claims = [];
+  const connections = await Promise.all(spec.connections.map(async cs => {
+    const directionOrdinal = DirectionEnum.values[cs.direction.replace(/ /g, '_').toUpperCase()];
+    if (directionOrdinal === undefined) {
+      throw Error(`Handle connection direction ${cs.direction} is not supported`);
+    }
+    const proto = {
+      name: cs.name,
+      direction: directionOrdinal,
+      type: await typeToProtoPayload(cs.type)
+    };
+    const claimsProto = claimsToProtoPayload(cs, proto);
+    if (claimsProto != null) {
+      claims = claims.concat(claimsProto);
+    }
+    return proto;
+  }));
   return {
     name: spec.name,
     location: spec.implFile,
-    connections: await Promise.all(spec.connections.map(async cs => {
-      const directionOrdinal = DirectionEnum.values[cs.direction.replace(/ /g, '_').toUpperCase()];
-      if (directionOrdinal === undefined) {
-        throw Error(`Handle connection direction ${cs.direction} is not supported`);
-      }
-      return {
-        name: cs.name,
-        direction: directionOrdinal,
-        type: await typeToProtoPayload(cs.type)
-      };
-    }))
+    connections,
+    claims
   };
 }
 
+// Converts the claims in HandleConnectionSpec.
+function claimsToProtoPayload(cs: HandleConnectionSpec, proto: {}) {
+  return cs.claims?.map(claim => {
+    const accessPath = {handleConnection: cs.name};
+    switch (claim.type) {
+      case ClaimType.IsTag: {
+        const tag = {semanticTag: claim.tag};
+        let predicate;
+        if (claim.isNot) {
+          predicate = {
+            not: {
+              predicate: {
+                literal: {label: tag}
+              }
+            }
+          };
+        } else {
+          predicate = {
+            literal: {label: tag}
+          };
+        }
```

Comment:
This might be a little shorter and sweeter:

```
let predicate = {literal: {label: tag}};
if (claim.isNot) {
  predicate = {not: {predicate}}
}
```

Reply:
That is what I tried initially, but got the following error:
```
src/tools/manifest2proto.ts(95,24): error TS2322: Type '{ not: { predicate: { literal: { label: { semanticTag: string; }; }; }; }; }' is not assignable to type '{ literal: { label: { semanticTag: string; }; }; \
}'.
Object literal may only specify known properties, and 'not' does not exist in type '{ literal: { label: { semanticTag: string; }; }; }'.
```

I believe it is related to the other question about importing `ClaimProto` type.

Reply:
I think it's because the type is inferred from the assignment in the `let` statement.

If you specify a type yourself, it will stop doing the incorrect inference:

```
let predicate: {} = {literal: {label: tag}};
```